torch/fx/experimental/symbolic_shapes.py

```diff
@@ -432,20 +432,31 @@ def _reduce_to_lowest_terms(expr: sympy.Expr) -> sympy.Expr:
     Useful when an expression is == or != to 0.
     """
     def integer_coefficient(x):
-        if isinstance(x, sympy.Integer):
+        if x.is_Integer:
             return abs(int(x))
-        elif isinstance(x, sympy.Mul):
-            return math.prod([abs(int(arg)) for arg in x.args if isinstance(arg, sympy.Integer)])
+        elif x.is_Mul:
+            return abs(int(x.args[0])) if x.args[0].is_Integer else 1
```

Comment:
I guess your point here is that if there are multiple integral factors they're guaranteed to have already been folded

Reply:
Yep. Added a comment mentioning this.